homeassistant/components/dwd_weather_warnings/sensor.py

```diff
@@ -9,137 +9,174 @@
-        ): vol.All(cv.ensure_list, [vol.In(MONITORED_CONDITIONS)]),
-    }
+from .const import (
+    API_ATTR_WARNING_COLOR,
+    API_ATTR_WARNING_DESCRIPTION,
+    API_ATTR_WARNING_END,
+    API_ATTR_WARNING_HEADLINE,
+    API_ATTR_WARNING_INSTRUCTION,
+    API_ATTR_WARNING_LEVEL,
+    API_ATTR_WARNING_NAME,
+    API_ATTR_WARNING_PARAMETERS,
+    API_ATTR_WARNING_START,
+    API_ATTR_WARNING_TYPE,
+    ATTR_LAST_UPDATE,
+    ATTR_REGION_ID,
+    ATTR_REGION_NAME,
+    ATTR_WARNING_COUNT,
+    ATTRIBUTION,
+    CONF_GPS_ACTIVE,
+    CONF_GPS_TRACKER,
+    CURRENT_WARNING_SENSOR,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    LOGGER,
+    SENSOR_TYPES,
 )
+from .util import get_position_data
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
     add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Set up the DWD-Weather-Warnings sensor."""
-    name = config.get(CONF_NAME)
-    region_name = config.get(CONF_REGION_NAME)
+    """Import the configurations from YAML to config flows."""
+    # Show issue as long as the YAML configuration exists.
+    async_create_issue(
+        hass,
+        DOMAIN,
+        "deprecated_yaml",
+        is_fixable=False,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_yaml",
+    )
+
+    # Only import the deprecated config, if it hasn't been done yet.
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.source == SOURCE_IMPORT:
+            return
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+        )
+    )
+
 
-    api = WrappedDwDWWAPI(DwdWeatherWarningsAPI(region_name))
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up entities from config entry."""
+    name: str = entry.data[CONF_NAME]
+    monitored_conditions: list[str] = entry.data[CONF_MONITORED_CONDITIONS]
+    api = hass.data[DOMAIN][entry.entry_id]
+    coordinator = DwdWeatherWarningsCoordinator(hass, entry, api, name)
 
-    sensors = [
-        DwdWeatherWarningsSensor(api, name, description)
-        for description in SENSOR_TYPES
-        if description.key in config[CONF_MONITORED_CONDITIONS]
-    ]
+    # Add sensor for every monitored condition.
+    entities: list[DwdWeatherWarningsSensor] = []
+    for description in SENSOR_TYPES:
+        if description.key in monitored_conditions:
+            entities.append(DwdWeatherWarningsSensor(coordinator, description, name))
 
-    add_entities(sensors, True)
+    async_add_entities(entities, True)
 
 
-class DwdWeatherWarningsSensor(SensorEntity):
-    """Representation of a DWD-Weather-Warnings sensor."""
+class DwdWeatherWarningsCoordinator(DataUpdateCoordinator):
+    """Custom coordinator for the dwd_weather_warnings integration."""
```

Comment:
We normally don't place coordinators in the platform files (sensor, switch, ...) since they are expected to be shared. Place it in __init__.py or coordinator.py